src/mon/MonMap.cc

```diff
@@ -383,6 +383,9 @@ void MonMap::print(ostream& out) const
   if (stretch_mode_enabled) {
     out << "stretch_mode_enabled " << stretch_mode_enabled << "\n";
     out << "tiebreaker_mon " << tiebreaker_mon << "\n";
+  }
+  if (stretch_mode_enabled ||
+      !disallowed_leaders.empty()) {
     out << "disallowed_leaders " << disallowed_leaders << "\n";
```

Comment:
Tested with master:
```
epoch 4
fsid da4afde7-5303-4864-b532-4f111e494c2b
last_changed 2021-11-15T20:53:22.777200+0000
created 2021-11-15T20:51:56.616368+0000
min_mon_release 17 (quincy)
election_strategy: 1
0: [v2:127.0.0.1:40087/0,v1:127.0.0.1:40088/0] mon.a
1: [v2:127.0.0.1:40089/0,v1:127.0.0.1:40090/0] mon.b
2: [v2:127.0.0.1:40091/0,v1:127.0.0.1:40092/0] mon.c
dumped monmap epoch 4
```
Tested on your branch:
```
epoch 4
fsid f85dab66-a13b-43d2-8c0d-d38d81418cda
last_changed 2021-11-15T21:01:45.555050+0000
created 2021-11-15T21:00:31.427033+0000
min_mon_release 17 (quincy)
election_strategy: 1
disallowed_leaders a
0: [v2:127.0.0.1:40463/0,v1:127.0.0.1:40464/0] mon.a
1: [v2:127.0.0.1:40465/0,v1:127.0.0.1:40466/0] mon.b
2: [v2:127.0.0.1:40467/0,v1:127.0.0.1:40468/0] mon.c
dumped monmap epoch 4
```
If I'm being nit-picky, it looks like you need to add a `:` after `disallowed_leaders`. But otherwise, the change works as it should.